src/textual/strip.py

```diff
@@ -157,6 +158,101 @@ def from_lines(
         """
         return [cls(segments, cell_length) for segments in lines]
 
+    @classmethod
+    def align(
+        cls,
+        strips: list[Strip],
+        style: Style,
+        width: int,
+        height: int,
+        horizontal: AlignHorizontal,
+        vertical: AlignVertical,
+    ) -> Iterable[Strip]:
+        """Align a list of strips on both axis.
+
+        Args:
+            strips: A list of strips, such as from a render.
+            style: The Rich style of additional space.
+            width: Width of container.
+            height: Height of container.
+            horizontal: Horizontal alignment method.
+            vertical: Vertical alignment method.
+
+        Returns:
+            An iterable of strips, with additional padding.
+
+        """
+        if not strips:
+            return
+        line_lengths = [strip.cell_length for strip in strips]
+        shape_width = max(line_lengths)
+        shape_height = len(line_lengths)
+
+        def blank_lines(count: int) -> Iterable[Strip]:
+            """Create blank lines.
+
+            Args:
+                count: Desired number of blank lines.
+
+            Returns:
+                An iterable of blank lines.
+            """
+            blank = cls([Segment(" " * width, style)], width)
+            for _ in range(count):
+                yield blank
+
+        top_blank_lines = bottom_blank_lines = 0
+        vertical_excess_space = max(0, height - shape_height)
+
+        if vertical == "top":
+            bottom_blank_lines = vertical_excess_space
+        elif vertical == "middle":
+            top_blank_lines = vertical_excess_space // 2
+            bottom_blank_lines = vertical_excess_space - top_blank_lines
+        elif vertical == "bottom":
+            top_blank_lines = vertical_excess_space
+
+        if top_blank_lines:
+            yield from blank_lines(top_blank_lines)
+
+        if horizontal == "left":
+            for strip in strips:
+                if strip.cell_length == width:
+                    yield strip
+                else:
+                    yield Strip(
+                        line_pad(strip._segments, 0, width - strip.cell_length, style),
+                        width,
+                    )
```

Comment:
This code looks like there's an assumption that the cell length of the strip is less than the width. Is this accounted for elsewhere?